public/app/plugins/panel/timeseries/plugins/ContextMenuPlugin.tsx

```diff
@@ -39,108 +38,37 @@ export const ContextMenuPlugin = ({
   replaceVariables,
   ...otherProps
 }: ContextMenuPluginProps) => {
-  const plotCanvas = useRef<HTMLDivElement>();
   const [coords, setCoords] = useState<ContextMenuSelectionCoords | null>(null);
   const [point, setPoint] = useState<ContextMenuSelectionPoint | null>(null);
   const [isOpen, setIsOpen] = useState(false);
 
-  const openMenu = useCallback(() => {
-    setIsOpen(true);
-  }, [setIsOpen]);
-
-  const closeMenu = useCallback(() => {
-    setIsOpen(false);
-  }, [setIsOpen]);
-
-  const clearSelection = useCallback(() => {
-    pluginLog('ContextMenuPlugin', false, 'clearing click selection');
-    setPoint(null);
-  }, [setPoint]);
-
-  // Add uPlot hooks to the config, or re-add when the config changed
   useLayoutEffect(() => {
-    let bbox: DOMRect | undefined = undefined;
-
-    const onMouseCapture = (e: MouseEvent) => {
-      let update = {
-        viewport: {
-          x: e.clientX,
-          y: e.clientY,
-        },
-        plotCanvas: {
-          x: 0,
-          y: 0,
-        },
-      };
-      if (bbox) {
-        update = {
-          ...update,
-          plotCanvas: {
-            x: e.clientX - bbox.left,
-            y: e.clientY - bbox.top,
-          },
-        };
-      }
-      setCoords(update);
-    };
-
-    // cache uPlot plotting area bounding box
-    config.addHook('syncRect', (u, rect) => {
-      bbox = rect;
-    });
+    let seriesIdx: number | null = null;
 
     config.addHook('init', (u) => {
-      const canvas = u.over;
-      plotCanvas.current = canvas || undefined;
-      plotCanvas.current?.addEventListener('mousedown', onMouseCapture);
-
-      pluginLog('ContextMenuPlugin', false, 'init');
-      // for naive click&drag check
-      let isClick = false;
-
-      // REF: https://github.com/leeoniya/uPlot/issues/239
-      let pts = Array.from(u.root.querySelectorAll<HTMLDivElement>('.u-cursor-pt'));
-
-      plotCanvas.current?.addEventListener('mousedown', () => {
-        isClick = true;
-      });
-
-      plotCanvas.current?.addEventListener('mousemove', () => {
-        isClick = false;
-      });
-
-      // TODO: remove listeners on unmount
-      plotCanvas.current?.addEventListener('mouseup', (e: MouseEvent) => {
-        // ignore cmd+click, this is handled by annotation editor
-        if (!isClick || e.metaKey || e.ctrlKey) {
-          setPoint(null);
-          return;
-        }
-        isClick = true;
-
-        if (e.target instanceof HTMLElement) {
-          if (!e.target.classList.contains('u-cursor-pt')) {
-            pluginLog('ContextMenuPlugin', false, 'canvas click');
-            setPoint({ seriesIdx: null, dataIdx: null });
-          }
+      u.over.addEventListener('click', (e) => {
```

Comment:
uplot now has some internal quality-of-life assistance that prevents clicks from firing when it detects drags (via a capturing listener on `.u-over`)